plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -875,4 +880,190 @@ CompletableFuture<GetObjectAttributesResponse> getBlobMetadata(S3AsyncClient s3A
 
         return SocketAccess.doPrivileged(() -> s3AsyncClient.getObjectAttributes(getObjectAttributesRequest));
     }
+
+    @Override
+    public void deleteAsync(ActionListener<DeleteResult> completionListener) {
+        try (AmazonAsyncS3Reference asyncClientReference = blobStore.asyncClientReference()) {
+            S3AsyncClient s3AsyncClient = asyncClientReference.get().client();
+
```

Comment:
Can this contend with the critical upload connections and queues?

Reply:
This uses the normal (low priortity) client. There are 2 other clients - priorityClient and urgentClient. This should not be causing contention to other important operations.

Reply:
Yes @vikasvb90. Explaining the reason below -

The method uses a CompletableFuture called listingFuture to represent the entire listing and deletion process. The deletion requests are executed as part of a chain of CompletableFuture objects, where each subsequent deletion batch is executed only after the previous batch has completed.

This chaining is achieved through the executeDeleteChain method, which takes the current CompletableFuture chain (currentChain) and appends the next batch of deletions to it, creating a new CompletableFuture (newChain). The newChain is then used as the currentChain for the next batch of deletions.

By relying on this single CompletableFuture chain, the code ensures that only one deletion batch is executed at a time, effectively serializing the deletion requests. This serialization is enforced by the nature of CompletableFuture chaining, where each subsequent operation is executed only after the previous one has completed.

Therefore, from the code perspective, the use of a single CompletableFuture chain in the deleteAsync method is the reason why there is only one deletion request being processed at any given time during the asynchronous deletion process.